### PR TITLE
Added NLP: Basic text classification with JAX & Flax

### PR DESCRIPTION
**Pull request by vfdev-5**
cc @melissawm to review

### CODE REVIEW

**Comment by jakevdp** [comment]
Looks really great – thanks for this! I'll wait on a review from @melissawm before merging.

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:17]:
```suggestion
In this tutorial we learn how to perform text classification from raw text data and train a basic 1D Convnet to perform sentiment analysis using JAX. This tutorial is originally inspired by ["Text classification from scratch with Keras"](https://keras.io/examples/nlp/text_classification_from_scratch/#build-a-model).
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:19]:
```suggestion
We will use the IMDB movie review dataset to classify the review to "positive" and "negative" classes. We implement from scratch a simple model using Flax, train it and compute metrics on the test set.
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:28]: nit:

```suggestion
- [tqdm](https://tqdm.github.io/) for a progress bar in the training step
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:61]:
```suggestion
Next, we will:
- create the dataset Python class to read samples from the disk;
- use [Tiktoken](https://github.com/openai/tiktoken) to encode raw text into tokens; and
- use [Grain](https://github.com/google/grain) for efficient data loading and batching.
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:103]: This reads better to me, but it's only a suggestion:
```suggestion
Now, we can create a string-to-tokens preprocessing transformation and set up data loaders. We are going to use the GPT-2 tokenizer via [Tiktoken](https://github.com/openai/tiktoken).
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:132]: If you are using two separate code cells for these actions, I would keep the grain import in the next cell so they are "self-contained" (they are not really self-contained, but it makes for a better cognitive unit)

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:193]: Can you leave a tiny note about why we are printing this, and what we should expect to see?

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:198]:
```suggestion
We choose a simple 1D convnet to classify the text. The first layer of the model transforms input tokens into float features using an embedding layer (`nnx.Embed`), then they are encoded further with convolutions. Finally, we classify encoded features using fully-connected layers.
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:273]: Why is this shape informative here?

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:404]: Just a suggestion:
```suggestion
Now, we can start the training. Note that this step can take quite a while to run, depending on the number of epochs you are using. 
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:412]:
```suggestion
Let's visualize the collected metrics:
```

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [inline docs/JAX_basic_text_classification.md:452]: Is it possible to include a quick summary/interpretation of what we are seeing in these labels?

**Review by melissawm — COMMENTED** [review]
Left a couple of optional suggestions, some to improve the text flow and some to include more insight into results/prints of the results. Cheers!

File: docs/JAX_basic_text_classification.md
Comment (vfdev-5) [reply docs/JAX_basic_text_classification.md:273]: To ensure that we output expected shape: (N, num_classes), updated to
```
print("Prediction shape (N, num_classes): ", y.shape)
```

File: docs/JAX_basic_text_classification.md
Comment (vfdev-5) [reply docs/JAX_basic_text_classification.md:404]: Well, this is not exactly true, in this case, whole training takes 2 minutes. The model and the dataset are rather small.

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [reply docs/JAX_basic_text_classification.md:28]: Just ignore then, thanks :)

File: docs/JAX_basic_text_classification.md
Comment (melissawm) [reply docs/JAX_basic_text_classification.md:404]: Interesting! For me it took something like 40 mins in colab.

File: docs/JAX_basic_text_classification.md
Comment (jakevdp) [inline docs/JAX_basic_text_classification.md:15]: Let's say "1D Convnet for basic text classification"

File: docs/JAX_basic_text_classification.md
Comment (jakevdp) [inline docs/JAX_basic_text_classification.md:476]: Remove empty cell

File: docs/JAX_basic_text_classification.md
Comment (jakevdp) [inline docs/JAX_basic_text_classification.md:474]: It would be nice to have some sort of conclusion text here – summarize what we did and what the next steps may be for someone interested in going deeper.

**Review by jakevdp — COMMENTED** [review]
Looks great, thanks!

File: docs/JAX_basic_text_classification.md
Comment (vfdev-5) [reply docs/JAX_basic_text_classification.md:474]: Yes, I wanted to add links on other NLP tutorials doing text classification with transformers once we have them.
I'll add a short summary on what we did.

**Review by trallard — APPROVED** [review]
This seems all ready to merge after a couple of rounds of reviews cc/ @jakevdp 🚀

**Review by jakevdp — APPROVED** [review]

**Pull request merged**
